File: C-Sharp/Code Challenges/QueueWithStacks/QueueWithStacks/QueueWithStacks/PseudoQueue.cs
using System;
using System.Collections.Generic;
using System.Text;
using StacksAndQueues;

namespace QueueWithStacks
{
    public class PseudoQueue
    {
        public Stack stack { get; set; }
        public PseudoQueue()
        {
            stack = new Stack();
        }

        public void Enqueue(int value)
        {
            stack.Push(value);
        }

        public int Dequeue()
        {
            Stack flipped = new Stack();
            while (stack.Top != null)
            {
                flipped.Push(stack.Pop());
            }
            int value = flipped.Pop();
            while (flipped.Top != null)
            {
                stack.Push(flipped.Pop());
            }
            return value;
        }
    }
}

File: Java/Data Structures and Algorithms/dsas/StacksAndQueues/src/main/java/stacksandqueues/PseudoQueue.java
package stacksandqueues;

public class PseudoQueue<T> {
    public Stack<T> stack = new Stack();

    public void enqueue(T value) {
        stack.push(value);
    }

    public T dequeue() {
        Stack<T> flipped = new Stack();
        while (stack.peek() != null) {
            flipped.push(stack.pop());
        }
        T value = flipped.pop();
        while (flipped.peek() != null) {
            stack.push(flipped.pop());
        }
        return value;
    }
}
